### ISSUE TITLE
Introduce an option to install Functions in Crossplane chart

### ISSUE THREAD

**Issue by aerfio** [crossplane/crossplane#5541]
### What problem are you facing?

I want to install a Function, e.g [function-patch-and-transform](https://marketplace.upbound.io/functions/crossplane-contrib/function-patch-and-transform/v0.4.0), as a part of Helm release. Documentation mentions how to do it for Providers [here](https://docs.crossplane.io/latest/concepts/providers/#install-with-helm), and also for Configurations: [link](https://docs.crossplane.io/latest/concepts/packages/#install-with-helm). However I don't see any option to do it for Functions. Docs state that [Functions are Crossplane Packages](https://docs.crossplane.io/latest/concepts/composition-functions/#install-a-composition-function), which made me think that maybe I can just install it via helm by passing a reference to function package as if it was a Configuration:

```bash
helm install crossplane \
crossplane-stable/crossplane \
--namespace crossplane-system \
--create-namespace \
--version 1.15.1 \
--set configuration.packages='{xpkg.upbound.io/crossplane-contrib/function-patch-and-transform:v0.4.0}'
```

However, it leads to this:
```yaml
apiVersion: pkg.crossplane.io/v1
kind: Configuration
metadata:
  creationTimestamp: "2024-04-03T13:35:08Z"
  generation: 1
  name: crossplane-contrib-function-patch-and-transform
  resourceVersion: "662"
  uid: 3987256a-eda0-46d2-886e-d8a42d8f0c22
spec:
  ignoreCrossplaneConstraints: false
  package: xpkg.upbound.io/crossplane-contrib/function-patch-and-transform:v0.4.0
  packagePullPolicy: IfNotPresent
  revisionActivationPolicy: Automatic
  revisionHistoryLimit: 1
  skipDependencyResolution: false
status:
  conditions:
  - lastTransitionTime: "2024-04-03T13:35:12Z"
    message: 'linting package contents failed: package meta type is not Configuration'
    reason: UnhealthyPackageRevision
    status: "False"
    type: Healthy
  - lastTransitionTime: "2024-04-03T13:35:09Z"
    reason: ActivePackageRevision
    status: "True"
    type: Installed
  currentIdentifier: xpkg.upbound.io/crossplane-contrib/function-patch-and-transform:v0.4.0
  currentRevision: crossplane-contrib-function-patch-and-transform-97397f24bd9f
```

So it's treated as a Configuration, which is correct behaviour imho, but I just tested this case to be sure.

I've also thought I could use `extraObjects` field in helm chart https://github.com/crossplane/crossplane/blob/d35ecef8b3d9094261dbda64e72498cda1520797/cluster/charts/crossplane/values.yaml#L170-L181 but with this `values.yaml` file:
```yaml
extraObjects:
- apiVersion: pkg.crossplane.io/v1beta1
  kind: Function
  metadata:
    name: function-patch-and-transform
    annotations:
      helm.sh/hook: post-install
  spec:
    package: xpkg.upbound.io/crossplane-contrib/function-patch-and-transform:v0.4.0
```

and this command:
```bash
helm install crossplane \
crossplane-stable/crossplane \
--namespace crossplane-system \
--create-namespace \
--version 1.15.1 \
-f values.yaml
```

it fails with:
```
Error: INSTALLATION FAILED: failed post-install: unable to build kubernetes object for deleting hook crossplane/templates/extra-objects.yaml: resource mapping not found for name: "function-patch-and-transform" namespace: "" from "": no matches for kind "Function" in version "pkg.crossplane.io/v1beta1"
ensure CRDs are installed first
```

The resources annotated with [helm.sh/post-install hook](https://helm.sh/docs/topics/charts_hooks/#the-available-hooks) are evaluated immediately. It also means that the commented out example under the `extraObjects` would not work either IIUC.

### How could Crossplane help solve your problem?

If I didn't miss anything I would like to propose to add a functionality to install a function in [the same way as I'm installing Providers and Configurations](https://github.com/crossplane/crossplane/blob/d35ecef8b3d9094261dbda64e72498cda1520797/cluster/charts/crossplane/values.yaml#L46-L52). Without this feature I have to install Crossplane in 2 seperate steps, first install Crossplane chart itself and then apply yaml with desired Functions.

**Comment by jbw976** [comment]
I believe this was closed by #5586, although an earlier effort was started in #5539

**Comment by phisco** [comment]
ops, sorry @haarchri, missed your previous attempt 😅
